lib/krane/restart_task.rb

```diff
@@ -97,62 +104,135 @@ def tags(status, deployments)
       %W(namespace:#{@namespace} context:#{@context} status:#{status} deployments:#{deployments.to_a.length}})
     end
 
-    def identify_target_deployments(deployment_names, selector: nil)
-      if deployment_names.nil?
-        deployments = if selector.nil?
-          @logger.info("Configured to restart all deployments with the `#{ANNOTATION}` annotation")
-          apps_v1_kubeclient.get_deployments(namespace: @namespace)
+    def identify_target_workloads(deployment_names, statefulset_names, daemonset_names, selector: nil)
+      if deployment_names.blank? && statefulset_names.blank? && daemonset_names.blank?
```

Comment:
Curious why `blank?` vs `nil`? Are we passing empty strings in one of the paths? Looks like the default param in `run!` is an empty list now instead of nil?

Reply:
While the default arg is the empty list, I can't guarantee that someone won't explicitly pass `nil` at some point. Using `blank?` seems preferable over `empty?` or `nil?` since it will catch either case.